arch/arm/src/armv7-m/gnu/arm_lazyexception.S

```diff
@@ -192,7 +192,8 @@ exception_common:
 	 * here prohibits nested interrupts without some additional logic!
 	 */
 
-	setintstack	r2, r3
+	setintstack	r2, r3				/* SP = IRQ stack top */
+	sub		sp, #8					/* Adjust SP (minus 8-byte offset) */
```

Comment:
why need sub 8 here?

Reply:
I thought I added a comment on this.
The reason is that the setinstack load $SP from g_intstackbase

```
#if !defined(CONFIG_SMP) && CONFIG_ARCH_INTERRUPTSTACK > 7                                                                                                                                         
    .macro  setintstack, tmp1, tmp2                                                                                                                                                                
    ldr     sp, =g_intstackbase                                                                                                                                                                    
    .endm                                                                                                                                                                                          
#endif        
````

Reply:
Yes, but it's right to let sp point to the pass of the end stack, because AAPCS require all stack push must substract sp first before store the value to stack.

Reply:
OK

Reply:
Done